### ISSUE TITLE
[java] UnusedPrivateMethod false positive: Autoboxing into Number

### ISSUE THREAD

**Issue by enexusde** [pmd/pmd#4625]
**Affects PMD Version:** pmd:3.19.0:check

**Rule:** [UnusedPrivateMethod](https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#unusedprivatemethod)

**Description:**
Calling a private method that accept one parameter of type Number (superclass of autoboxing-type Integer)
**Code Sample demonstrating the issue:**

```java
		// ...
		foo(2);
	}

	private void foo(Number bar) {
		System.out.println(bar);
	}
```

**Expected outcome:**

PMD reports a violation at line 124, but that's wrong. That's a false positive.

**Running PMD through:** *Maven*

**Workaround:**

Cast the parameter to Number will work:
```java
		// ...
		foo((Number) 2);
	}

	private void foo(Number bar) {
		System.out.println(bar);
	}
```
